optax/contrib/_sam.py

```diff
@@ -81,7 +83,7 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
-@chex.dataclass
+@dataclasses.dataclass
```

Comment:
This is needed for tests to pass since chex.dataclass is incompatible with the newest JAX.